src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java

```diff
@@ -521,29 +522,39 @@ public OutputStream getOutputStream() throws UnsupportedDataAccessOperationExcep
 
     @Override
     public InputStream getAuxFileAsInputStream(String auxItemTag) throws IOException {
-        open();
         String destinationKey = getDestinationKey(auxItemTag);
         try {
-            if (this.isAuxObjectCached(auxItemTag)) {
-                S3Object s3object = s3.getObject(new GetObjectRequest(bucketName, destinationKey));
-                return s3object.getObjectContent();
-            } else {
-                logger.fine("S3AccessIO: Aux object is not cached, unable to get aux object as input stream.");
-                return null;
-            }
+            S3Object s3object = s3.getObject(new GetObjectRequest(bucketName, destinationKey));
+            return s3object.getObjectContent();
         } catch (AmazonClientException ase) {
-            logger.warning("Caught an AmazonServiceException in S3AccessIO.getAuxFileAsInputStream():    " + ase.getMessage());
-            throw new IOException("S3AccessIO: Failed to get aux file as input stream");
+            logger.fine("Caught an AmazonServiceException in S3AccessIO.getAuxFileAsInputStream() (object not cached?):    " + ase.getMessage());
+            return null;
         }
     }
 
     private String getDestinationKey(String auxItemTag) throws IOException {
         if (dvObject instanceof DataFile) {
+            if (key == null) {
+                String baseKey = this.getDataFile().getOwner().getAuthority() + "/" + this.getDataFile().getOwner().getIdentifier();
+                String storageIdentifier = dvObject.getStorageIdentifier();
+
+                if (storageIdentifier == null || "".equals(storageIdentifier)) {
+                    throw new FileNotFoundException("Data Access: No local storage identifier defined for this datafile.");
+                }
+
+                if (storageIdentifier.startsWith("s3://")) {
+                    bucketName = storageIdentifier.substring(storageIdentifier.indexOf(":") + 3, storageIdentifier.lastIndexOf(":"));
```

Comment:
If you are going to have a hard +3, why not just assume that the index will be 6?

Reply:
I copy-and-pasted this fragment from somewhere else in the class. And I'm assuming it was originally copy-and-pasted from another place, where this piece of code was run on storage identifiers with different prefixes (not just "s3:", but also "file: and "swift:", etc.). So the idea was that the length of "://" is always 3; but the length of "{prefix}://" can be anything. 
But yeah, the real crime with the fragment above is that it's now duplicated in 2 places in the class; I was just trying to quickly make it possible to read aux files without doing the full .open() on the object. I'll clean it up.